Essay_Analysis_Tool/Business/MainViewController.cs

```diff
@@ -46,10 +47,11 @@ public class MainViewController
         #endregion
 
         #region Constructor
-        public MainViewController(IMainView view)
+
+        public MainViewController(IMainView view, LoggerViewController loggerViewController)
         {
             _view = view;
-            _view.SetController(this);
+            _view.SetController(this, loggerViewController);
```

Comment:
This is a problem I was having with this architecture. I was not sure how I could access the logger's logging functionlaity from the controller of the MainView. I was not sure what the proper way of doing this was (even though I have done this very thing at work before). I could not remember what the best practice was, if you have a suggestion to solve this issue, I would be open to changing the code for it.

Reply:
It looks like what we should do here is any functionality that needs to be accessed by multiple controllers should be extracted to a service class and injected into the controllers (this would have to be done with a custom controller base class, not the one .NET provides).

Reply:
It also seems that the controllers were placed into the business layer which is incorrect. This is bad on my part, I will fix this and create the abstracted service class as soon as I can.

Reply:
Yes. Services should be injected to the Controllers and also a suggestion for the Views:

``` c#
class View : Form
{
    private readonly ViewController _controller;

    public View(ViewController controller)
    {
        _controller = controller;
    }
}

class ViewController
{
    private readonly View _view;

    public ViewController()
    {
        _view = new View(this);
    }
}
```

Reply:
Thank you, I was pulling from an example that I found online and I had some suspicions that it wasn't correct. Damn my American education.

Reply:
I think interfaces for the Views is overkill because i don't think there will be more than one view for one Controller or You want to plan multiple View?

Reply:
I really only want to have interfaces for the views because A.) Yes we should be prepared for multiple views and B.) The controllers do not need to understand the definition of those methods.

Reply:
I don't remember where i read but in a (probably) StackOverflow answer somebody advised that don't try to code for the far future only solve the current problem(s). This would allow us to write less code which would help to maintain the project more easily without unnecessary abstraction.

I understand Your second point but i only use interfaces when there is two or more implementation of the same abstract logic i do not use them when i just try to separate layers from each other. I will refactor my code to interfaces when there is a valid use-case of course but not until than.

Reply:
I have removed the interfaces as you were right, they were overkill.